lib/chef/provider/package/aix.rb

```diff
@@ -88,19 +88,24 @@ def load_current_resource
 
         def candidate_version
           return @candidate_version if @candidate_version
-          ret = shell_out_with_timeout("installp -L -d #{@new_resource.source}")
-          ret.stdout.each_line do |line|
-            case line
-            when /\w:#{Regexp.escape(@new_resource.package_name)}:(.*)/
-              fields = line.split(":")
-              @candidate_version = fields[2]
-              @new_resource.version(fields[2])
-              Chef::Log.debug("#{@new_resource} setting install candidate version to #{@candidate_version}")
+          if @new_resource.source
+            @package_source_found = ::File.exists?(@new_resource.source)
+            if @package_source_found
```

Comment:
yeah please make that fix, and change the code in load_current_resource and define_resource_requirements as well, it'll make it a lot easier to understand what the problem is and that it is being solved correctly.  i think last time we reviewed this i had issues reading the code and understanding how the problem mapped to the code.